### PR TITLE
DOCSP-43081: Change Streams

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-43081>
Staging - <https://deploy-preview-37--docs-c.netlify.app/read/change-streams>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/read/change-streams.txt [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+Open a Change Stream
+--------------------
+
+To open a change stream, call one of the following functions that corresponds to the
+scope of events you want to observe.
```
Comment (jordan-smith721) [inline source/read/change-streams.txt:40]:
```suggestion
scope of events you want to observe:
```

File: source/read/change-streams.txt [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+- ``mongoc_client_watch()``: To monitor all changes in the MongoDB deployment
+- ``mongoc_database_watch()``: To monitor changes in all collections in the database
+- ``mongoc_collection_watch()``: To monitor changes in the collection
```
Comment (jordan-smith721) [inline source/read/change-streams.txt:44]:
```suggestion
- ``mongoc_client_watch()``: Monitors all changes in the MongoDB deployment
- ``mongoc_database_watch()``: Monitors changes in all collections in the database
- ``mongoc_collection_watch()``: Monitors changes in the collection
```

File: source/read/change-streams.txt [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+     ...
+   }
+
+Modify the Change Stream Output
+-------------------------------
+
+You can pass the ``pipeline`` parameter to any watch function to modify the
+change stream output. This parameter allows you to watch for only specified
+change events. Format the parameter as a list of objects that each represents an
+aggregation stage.
```
Comment (jordan-smith721) [inline source/read/change-streams.txt:102]: S:
```suggestion
change events. Format the parameter as a list of objects, where each object represents an
aggregation stage.
```

File: source/read/change-streams.txt [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+- ``$addFields``
+- ``$match``
+- ``$project``
+- ``$replaceRoot``
+- ``$replaceWith``
+- ``$redact``
+- ``$set``
+- ``$unset``
+
+Match Specific events Example
```
Comment (jordan-smith721) [inline source/read/change-streams.txt:115]:
```suggestion
Match Specific Events Example
```

File: source/read/change-streams.txt [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example uses the ``pipeline`` parameter to include a ``$match`` stage
+to open a change stream that only records update operations:
```
Comment (jordan-smith721) [inline source/read/change-streams.txt:119]:
```suggestion
to open a change stream that records only update operations:
```

**Review by jordan-smith721 — CHANGES_REQUESTED** [review]
Nice job! Just a few changes

File: source/read/change-streams.txt [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+     }
+     ...
+   }
+
+Modify the Change Stream Output
+-------------------------------
+
+You can pass the ``pipeline`` parameter to any watch function to modify the
+change stream output. This parameter allows you to watch for only specified
+change events. Format the parameter as a list of object, where each object represents an
```
Comment (jordan-smith721) [inline source/read/change-streams.txt:101]:
```suggestion
change events. Format the parameter as a list of objects, where each object represents an
```

**Review by jordan-smith721 — APPROVED** [review]
LGTM w a small fix!

File: source/includes/read/change-streams.c [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        // start-open-change-stream
+        bson_t *pipeline = bson_new ();
+        // Change stream will wait 1 minute for changes
+        bson_t *opts = BCON_NEW ("maxAwaitTimeMS", BCON_INT64 (60000));
+        const bson_t *doc;
+
+        mongoc_change_stream_t *change_stream =
+            mongoc_collection_watch (collection, pipeline, opts);
+
+        while (mongoc_change_stream_next (change_stream, &doc)) {
```
Comment (kevinAlbs) [inline source/includes/read/change-streams.c:27]: `mongoc_change_stream_next` only returns true when a change is returned within the `maxAwaitTimeMS`. If no change was returned, and no error occurred, a later call to `mongoc_change_stream_next` can still return a change.

If the examples are intended to show the behavior of "wait (possibly forever) until there is an event or error", then I suggest looping as follows:

```c
while (true)
{
  bson_error_t error;
  if (mongoc_change_stream_next(change_stream, &doc))
  {
    char *str = bson_as_canonical_extended_json(doc, NULL);
    printf("Received change: %s\n", str);
    bson_free(str);
  }
  else if (mongoc_change_stream_error_document(change_stream, &error, NULL))
  {
    printf("Got error on change stream: %s\n", error.message);
    break;
  }
}
```

**Review by kevinAlbs — APPROVED** [review]
LGTM as-is, but added a comment to consider an alternative looping.

**Pull request merged**